pkg/gui/view_helpers.go

```diff
@@ -148,3 +148,45 @@ func (gui *Gui) postRefreshUpdate(c types.Context) error {
 
 	return nil
 }
+
+type inlineStatusInfo struct {
+	refCount int
+	stop     chan struct{}
+}
+
+func (gui *Gui) startRenderingInlineStatus(contextKey types.ContextKey) {
```

Comment:
thoughts on  moving this stuff to a helper?

Reply:
Ha, I thought I could get away with just putting it into a source file that has "helper" in its name. 😄

Reply:
Alas, I still wanna shrink down the gui struct so that it's not so much of a 'God Struct', and the helpers in the helpers package are one way to do that, so I reckon we extract a helper out for this

Reply:
OK, I can give it a try. Just to be clear, I'd move everything that's now in StateAccessor to the helper, including the state, right?

Reply:
I was thinking moreso that you'd leave the state available in StateAccessor but move the functions specific to item operations to the helper

Reply:
Here's what I came up with: 206a4e0f5789. Do let me know if you want something to be done differently, I'm not tired of the review yet. 😄 

Also, normally I'd make an effort to squash this down somehow into earlier commits (also your itemOperations refactor) to avoid this back-and-forth in the history, but it's too much work this time, so I'd rather leave it as is.